File: back-end/src/main/java/com/csulb/cookie/domain/PrebuildCake.java
package com.csulb.cookie.domain;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.util.ArrayList;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;


@TableName(value ="prebuild_cake")
@AllArgsConstructor
@NoArgsConstructor
@Data
public class PrebuildCake implements Serializable {

    @TableId(type = IdType.AUTO)
    private Integer id;

    @TableField("m_id")
    private Integer merchantId;

    private String cakeName;

    @TableField("base_id")
    private Integer baseSizeId;

    private Integer baseColorId;

    private Integer baseFlavorId;

    private Integer frostingColorId;

    private Integer frostingFlavorId;

    private Double price;

    @TableField(exist = false)
    private ArrayList<Integer> toppingIds;

}

File: front-end/lib/domain/PrebuildCake.dart
class PrebuildCake {


  int merchantId;
  String cakeName;
  int baseSizeId;
  int baseFlavorId;
  int baseColorId;
  int frostingColorId;
  int frostingFlavorId;
  double price;
  List<int> toppingIds;

  PrebuildCake(

      this.merchantId,
      this.cakeName,
      this.baseSizeId,
      this.baseFlavorId,
      this.baseColorId,
      this.frostingColorId,
      this.frostingFlavorId,
      this.price,
      this.toppingIds);
}